lxc/storage_volume.go

```diff
@@ -363,8 +363,34 @@ func (c *cmdStorageVolumeCopy) Run(cmd *cobra.Command, args []string) error {
 	srcServer := srcResource.server
 	srcPath := srcResource.name
 
-	// If a target was specified, specify the volume on the given member.
-	if c.storage.flagTarget != "" {
+	// If the source server is standalone then --target cannot be provided.
+	if c.storage.flagTarget != "" && !srcServer.IsClustered() {
+		return fmt.Errorf(i18n.G("Cannot set --target when source server is not clustered"))
+	}
+
+	// Get source pool and volume name
+	srcVolName, srcVolPool := c.storageVolume.parseVolumeWithPool(srcPath)
+	if srcVolPool == "" {
+		return fmt.Errorf(i18n.G("No storage pool for source volume specified"))
+	}
+
+	// Check if requested storage volume exists.
+	srcVolParentName, srcVolSnapName, srcIsSnapshot := api.GetParentAndSnapshotName(srcVolName)
+	srcVol, _, err := srcServer.GetStoragePoolVolume(srcVolPool, "custom", srcVolParentName)
+	if err != nil {
+		return err
+	}
+
+	// If the volume is in local storage, set the target to its location (or provide a helpful error
+	// message if the target is incorrect). If the volume is in remote storage (and the source server is clustered) we
+	// can use any provided target.
+	if srcVol.Location != "" {
+		if c.storage.flagTarget != "" && c.storage.flagTarget != srcVol.Location {
```

Comment:
I'm wondering if we should move this check to the LXD server side so that API users will benefit from it too.
We could modify the `Storage volume not found` error to something like `Storage volume not found on this cluster member` if the server was clustered?  This would avoid the additional round trips and DB queries in the client too.

What do you think?

Reply:
I don't mind, but the `srcVol` is required in the calls to `MoveStoragePoolVolume` and `CopyStoragePoolVolume` anyway, all I've done here is move the call to `GetStoragePoolVolume` further up in order to perform this check.

Reply:
Right, in that case its not doing any extra queries, then that is fine. 
Generally though I do try and push validation into the LXD server so its done for both `lxc` and direct API clients.

Reply:
Yep that makes sense. I'll add something on the API side too :+1:

Reply:
Can do that in a separate PR